File: hu.bme.mit.incqueryd.runtime/hu.bme.mit.incqueryd.core/src/test/java/hu/bme/mit/incqueryd/core/rete/actors/tests/BetaActorTest.java
package hu.bme.mit.incqueryd.core.rete.actors.tests;

import hu.bme.mit.incqueryd.core.rete.actors.testkits.BetaActorTestKit;
import hu.bme.mit.incqueryd.core.rete.nodes.data.BetaTestData;
import hu.bme.mit.incqueryd.core.test.util.GsonParser;
import hu.bme.mit.incqueryd.core.test.util.TestCaseFinder;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;

import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;

import akka.actor.ActorSystem;

import com.google.gson.Gson;
import com.google.gson.JsonIOException;
import com.google.gson.JsonSyntaxException;

/**
 * Test cases for the beta nodes.
 * 
 * @author szarnyasg
 * 
 */
public class BetaActorTest {

	static ActorSystem system;

	@BeforeClass
	public static void setup() {
		system = ActorSystem.create();
	}

	@AfterClass
	public static void teardown() {
		system.shutdown();
	}

	private void betaNodeTest(final String typeString) throws FileNotFoundException, IOException {
		final String fileNamePrefix = typeString + "-test-";
		
		final File[] files = TestCaseFinder.getTestCases(fileNamePrefix + "*.json");

		for (final File testFile : files) {
			final String recipeFile = testFile.getPath().replace(fileNamePrefix, typeString + "-recipe-");
			final Gson gson = GsonParser.getGsonParser();
			final BetaTestData data = gson.fromJson(new FileReader(testFile), BetaTestData.class);

			final BetaActorTestKit testKit = new BetaActorTestKit(system, recipeFile);
			testKit.test(data);
		}
	}
	
	@Test
	public void joinNodeTest() throws JsonSyntaxException, JsonIOException, IOException {
		betaNodeTest("join");
	}

	@Test
	public void antiJoinNodeTest() throws JsonSyntaxException, JsonIOException, IOException {
		betaNodeTest("antijoin");
	}

}

File: hu.bme.mit.incqueryd.client/hu.bme.mit.incqueryd.rete.actors/src/test/scala/hu/bme/mit/incqueryd/engine/rete/actors/tests/BetaActorTest.scala
package hu.bme.mit.incqueryd.engine.rete.actors.tests

import java.io.FileReader
import org.junit.AfterClass
import org.junit.BeforeClass
import org.junit.Test
import BetaActorTest.system
import akka.actor.ActorSystem
import eu.mondo.utils.NetworkUtils
import hu.bme.mit.incqueryd.actorservice.AkkaUtils
import hu.bme.mit.incqueryd.actorservice.YarnActorService
import hu.bme.mit.incqueryd.engine.rete.actors.testkits.BetaActorTestKit
import hu.bme.mit.incqueryd.engine.rete.nodes.data.BetaTestData
import hu.bme.mit.incqueryd.engine.test.util.GsonParser
import hu.bme.mit.incqueryd.engine.test.util.TestCaseFinder
import org.junit.Before
import org.junit.After
import scala.concurrent.Await

object BetaActorTest {

  var system: ActorSystem = _

  @BeforeClass
  def setup() {
    system = AkkaUtils.getRemotingActorSystem(YarnActorService.actorSystemName, NetworkUtils.getLocalIpAddress, YarnActorService.port)
  }

  @AfterClass
  def teardown() {
     //Await.result(system.terminate(), AkkaUtils.defaultTimeout)
  }
}

class BetaActorTest {

  private def betaNodeTest(typeString: String) {
    val fileNamePrefix = typeString + "-test-"
    val files = TestCaseFinder.getTestCases(fileNamePrefix + "*.json")
    for (testFile <- files) {
      val recipeFile = testFile.getPath.replace(fileNamePrefix, typeString + "-recipe-")
      val gson = GsonParser.getGsonParser
      val data = gson.fromJson(new FileReader(testFile), classOf[BetaTestData])
      val testKit = new BetaActorTestKit(system, recipeFile)
      testKit.test(data)
    }
  }

  @Test
  def joinNodeTest() {
    betaNodeTest("join")
  }

  @Test
  def antiJoinNodeTest() {
    betaNodeTest("antijoin")
  }
}
